### ISSUE TITLE
WireOutputObject throws on write method

### ISSUE THREAD

**Issue by oety** [OpenHFT/Chronicle-Wire#108]
According to the javadoc: https://docs.oracle.com/javase/7/docs/api/java/io/DataOutput.html#write(int)

The write method should ignore the 24 high order bits, however in WriteOutputObject we get an illegal state exception if we pass any integer that has any of the higher bits set:

```java
@Test
    public void testWriteByte() throws IOException {
        final Bytes bytes = Bytes.elasticByteBuffer();
        try {
            final Wire wire = new BinaryWire(bytes);
            DataOutput out = new WireObjectOutput(wire);
            int b = 256;
            out.write(b); // expecting 0 to be written
        } catch (IllegalArgumentException e) {
            fail("should not throw");
        }
        finally {
            bytes.release();
        }
    }
```

**Comment by hft-team-city** [comment]
Released in Chronicle-Wire-2.17.5 (https://github.com/OpenHFT/Chronicle-Wire/releases/chronicle-wire-2.17.5), BOM-2.17.30 (https://github.com/OpenHFT/OpenHFT/releases/chronicle-bom-2.17.30)

**Comment by hft-team-city** [comment]
Released in [Chronicle-Wire-2.20.101](https://github.com/OpenHFT/Chronicle-Wire/releases/chronicle-wire-2.20.101), [BOM-2.20.134](https://github.com/OpenHFT/OpenHFT/releases/chronicle-bom-2.20.134)
